core/src/main/java/org/infinispan/cache/impl/DecoratedCache.java

```diff
@@ -116,22 +98,22 @@ public void stop() {
 
    @Override
    public boolean lock(K... keys) {
-      return cacheImplementation.lock(Arrays.asList(keys), flags, classLoader.get());
+      return cacheImplementation.lock(Arrays.asList(keys), flags, null);
```

Comment:
The methods with ClassLoader are package-private, therefore non-public API - can't you remove the parameter altogether?